subsys/bluetooth/host/hci_core.c

```diff
@@ -8983,6 +8983,11 @@ void bt_le_scan_cb_register(struct bt_le_scan_cb *cb)
 {
 	sys_slist_append(&scan_cbs, &cb->node);
 }
+
+void bt_le_scan_cb_unregister(struct bt_le_scan_cb *cb)
+{
+	sys_slist_find_and_remove(&scan_cbs, &cb->node);
```

Comment:
Is this actually safe from 2 different context? `SYS_SLIST_FOR_EACH_NODE_SAFE` has this warning ` * This and other SYS_SLIST_*() macros are not thread safe.` I cannot see why `SYS_SLIST_FOR_EACH_CONTAINER_SAFE` would be different.

Reply:
Hi @joerchan ,
Sorry for reply late,  
In face, this is not called from 2 different contexts, just considering whether users will delete themselves from the callback call, in order to avoid ambiguity, I will remove the SYS_SLIST_FOR_EACH_CONTAINER_SAFE changes, thanks for review.

Reply:
@anchao Sorry, I think I did not explain it properly. The _SAFE variant makes sense so that you can unregister the callback from the recv callback, I'm not objecting to this.

I'm asking if the cb list needs to be protected by a mutex in addition, because the unregister is an API call and can be called from any context. If it is called from a preemtible thread it might be interrupted in the middle of removing the callback from the list.

Reply:
> If it is called from a preemtible thread

Doesn't a lot of functions in the host API have this problem? I was under the impression that calls from preemptive threads weren't really supported?

@jhedberg is this the case?

Reply:
I suppose that is true.  In that case we can take that as a separate issue considering the entire BT API.

And add back the SYS_SLIST_FOR_EACH_CONTAINER_SAFE change. :)

Reply:
Thanks for comment, I brought back the SYS_SLIST_FOR_EACH_CONTAINER_SAFE, Please review again.